strawberry/http/base.py

```diff
@@ -0,0 +1,62 @@
+import json
+from typing import Any, Dict, Generic, List, Mapping, Optional, Union
+from typing_extensions import Protocol
+
+from strawberry.http import GraphQLHTTPResponse
+
+from .exceptions import HTTPException
+from .typevars import Request
+
+
+class BaseRequestProtocol(Protocol):
+    @property
+    def query_params(self) -> Mapping[str, Optional[Union[str, List[str]]]]:
+        ...
+
+    @property
+    def method(self) -> str:
```

Comment:
Maybe this can be an enum

Reply:
or a literal too :D

that might be quite nice as I won't need to use .lower then!